src/common/utils/fhir/formatter.ts

```diff
@@ -0,0 +1,48 @@
+import { ProductDto } from 'src/common/dtos/product.dto';
+import { SystemProductDto } from 'src/common/dtos/system-product.dto';
+import { R4 } from '@ahryman40k/ts-fhir-types';
+
+export const getSystems = (products: ProductDto[]): Array<any> => {
+  const mappings = products.map(product => product.mappings);
+  const m = [].concat.apply([], mappings);
```

Comment:
Looks like you could map over mappings to fetch the names instead of creating a separate container.

Reply:
I was trying to avoid nesting map... but yeah you are right, let me do the refactor